packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/CameraProperties.java

```diff
@@ -0,0 +1,166 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera;
+
+import android.graphics.Rect;
+import android.hardware.camera2.CameraAccessException;
+import android.hardware.camera2.CameraCharacteristics;
+import android.hardware.camera2.CameraManager;
+import android.os.Build.VERSION_CODES;
+import android.util.Range;
+import android.util.Rational;
+import android.util.Size;
+import androidx.annotation.RequiresApi;
+
+/** An interface allowing access to the different characteristics of the device's camera. */
+public interface CameraProperties {
+  String getCameraName();
```

Comment:
It's not obvious what most of these methods mean. Please add Javadoc comments to each public method (as required by style); it seems like the best option would probably be to document them by indicating what CameraCharacteristics entry each one maps to.